syntax = "proto3";

// Permissions service definitions used by OttServices.
// This file was moved to app-gateway2/app-gateway/OttServices to keep
// service definitions co-located with the plugin that consumes them.

package ottx.permission;

// AppKey identifies the application and partner.
message AppKey {
  // Application identifier (e.g., AppKey.app).
  string app = 1;

  // Syndication partner identifier (e.g., AppKey.syndication_partner_id).
  string syndication_partner_id = 2;
}

// Request for enumerating permissions for an app.
message EnumeratePermissionsRequest {
  // The application key (app ID + partner).
  AppKey app_key = 1;

  // Optional filters restricting which permissions are returned.
  repeated string permission_filters = 2;
}

// Response containing a list of permissions.
message EnumeratePermissionsResponse {
  // A collection of permission strings applicable to the app.
  repeated string permissions = 1;
}

// gRPC service for enumerating application permissions.
service AppPermissionsService {
  // Returns the list of permissions for the specified application.
  rpc EnumeratePermissions(EnumeratePermissionsRequest)
      returns (EnumeratePermissionsResponse);
}
